test/unit/parse_tree_visitor_test.rb

```diff
@@ -26,6 +26,27 @@ def test_dynamic_variable
     )
   end
 
+  def test_echo
+    assert_equal(
+      ["test"],
+      visit(%({% echo test %}))
+    )
+  end
+
+  def test_echo_with_filter
+    assert_equal(
+      ["test", "infilter"],
+      visit(%({% echo test | split: infilter %}))
+    )
+  end
+
+  def text_echo_dynamic_variable
+    assert_equal(
+      ["test", "inlookup"],
+      visit(%({% echo test[inlookup] %}))
+    )
+  end
```

Comment:
I think we can remove these, since they are basically testing the variable instead of testing anything new for the echo tag.

```suggestion
```